wagtail_transfer/operations.py

```diff
@@ -390,16 +391,41 @@ def _populate_fields(self, context):
             elif isinstance(field, StreamField):
                 value = json.dumps(update_object_ids(field.stream_block, json.loads(value), context.destination_ids_by_source))
 
+            elif isinstance(field, models.ManyToManyField):
+                # setting forward ManyToMany directly is prohibited
+                continue
+
             setattr(self.instance, field.get_attname(), value)
 
+    def _populate_many_to_many_fields(self, context):
+        save_needed = False
+
+        # for ManyToManyField, this must be done after saving so that the instance has an id.
+        # for ParentalManyToManyField, this could be done before, but doing both together avoids additional
+        # complexity as the method is identical
+        for field in self.model._meta.get_fields():
+            if isinstance(field, models.ManyToManyField):
+                try:
+                    value = self.object_data['fields'][field.name]
+                except KeyError:
+                    continue
+                target_model = get_base_model(field.related_model)
+                # translate list of source site ids to destination site ids
+                value = [context.destination_ids_by_source[(target_model, pk)] for pk in value]
+                getattr(self.instance, field.get_attname()).set(value)
+                save_needed = True
+        if save_needed:
+            # _save() for creating a page may attempt to re-add it as a child, so the instance (assumed to be already
+            # in the tree) is saved directly
+            self.instance.save()
```

Comment:
I was about to propose a plan for avoiding this second save() (regular M2M relations don't need it as the DB change happens immediately on write, and IIRC, ParentalM2M managers have a `commit` method that applies the change for that individual relation) - but on second thoughts, I think keeping it in is probably a good thing from the point of view of things like search indexing (so that the post_save signal gets called on the final updated version of the object).

Reply:
I saw that regular M2M didn't need it, but didn't find `commit` - will remember that one for next time! That makes sense though - I also just went with save for consistency.